client/components/get-directions.jsx

```diff
@@ -1,45 +1,55 @@
 import React from 'react';
 
-export default class RouteDetails extends React.Component {
+export default class getDirections extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       distance: '',
-      duration: ''
+      duration: '',
+      saved: false
     };
     this.mapRef = React.createRef();
     this.directionsPanelRef = React.createRef();
     this.mapInstance = null;
     this.directionsService = null;
     this.directionsRenderer = null;
+    this.handleClickSave = this.handleClickSave.bind(this);
   }
 
   componentDidMount() {
-    this.directionsService = new google.maps.DirectionsService();
-    this.directionsRenderer = new google.maps.DirectionsRenderer(
-      { draggable: true });
     this.mapInstance = new google.maps.Map(this.mapRef.current);
-    this.directionsRenderer.setMap(this.mapInstance);
-    this.directionsRenderer.setPanel(this.directionsPanelRef.current);
-    this.calcRoute();
+    this.directionsService = new google.maps.DirectionsService();
+    this.directionsRenderer = new google.maps.DirectionsRenderer({
+      draggable: true,
+      map: this.mapInstance,
+      panel: this.directionsPanelRef.current
+    });
+    this.directionsRenderer.addListener('directions_changed', () => {
+      this.calcTotals(this.directionsRenderer.getDirections());
+    });
+    this.displayRoute();
   }
 
-  calcRoute() {
-    const { A, B, C } = this.props.locations;
+  displayRoute() {
+    // const { A, B, C } = this.props.locations;
     const req = {
-      origin: { placeId: A.place_id },
-      destination: { placeId: A.place_id },
+      origin: { placeId: 'ChIJYVqRI4dskFQRVWnuu-Qjk0E' },
+      destination: { placeId: 'ChIJYVqRI4dskFQRVWnuu-Qjk0E' },
       waypoints: [
-        { location: { placeId: B.place_id } },
-        { location: { placeId: C.place_id } }
+        { location: { placeId: 'ChIJW4VXzodskFQRseSBkOHh9Vg' } },
+        { location: { placeId: 'ChIJF7Wdd4ZskFQRhk02OceIcdo' } }
       ],
+      // origin: { placeId: A.place_id },
+      // destination: { placeId: A.place_id },
+      // waypoints: [
+      //   { location: { placeId: B.place_id } },
+      //   { location: { placeId: C.place_id } }
+      // ],
```

Comment:
commented out code! also, did you mean to hard-code these place id's?
![sad cat](https://www.nydailynews.com/resizer/qsDdUADuJZhCgCrhGEBLXw4eLVs=/fit-in/800x533/smart/filters:fill(black)/arc-anglerfish-arc2-prod-tronc.s3.amazonaws.com/public/BRPREEN3Q25Y2VEMWKCRT5ZQJU.jpg)

Reply:
Darn, I need to remember to delete stuff when I'm done testing! 😅